### PR TITLE
Minicap issue# 54 android15 support

### PR DESCRIPTION
**Pull request by varundtsfi**
There is a following bug which was creating the problem to the people.
#54
#54 (comment)
Now Android 15 Beta suppprt added and verified. It's. working fine.
Screenshot from 2024-05-22 20-31-33

### CODE REVIEW

File: libs/arm64-v8a/gdb.setup [hunk 1]
```diff
@@ -0,0 +1,2 @@
+set solib-search-path ./obj/local/arm64-v8a
+directory /home/radmin/Downloads/android-ndk-r15c/sysroot /home/radmin/Downloads/android-ndk-r15c/sources/android/support /home/radmin/Downloads/android-ndk-r15c/sources/android/support/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++ /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/include /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/libjpeg-turbo-1.4.1/simd /home/radmin/Documents/Public_Git_Projects/minicap/jni/minicap-shared/aosp/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi/include jni/libpng
```
Comment (koral--) [inline libs/arm64-v8a/gdb.setup:2]: This file should not be committed. It contains paths specific to your machine.
The same applies to other gdb.setup files.

Comment (varundtsfi) [reply libs/arm64-v8a/gdb.setup:2]: Ok. let me remove the file

Comment (varundtsfi) [reply libs/arm64-v8a/gdb.setup]: FIle Removed

File: libs/armeabi-v7a/gdb.setup [hunk 2]
```diff
@@ -0,0 +1,2 @@
+set solib-search-path ./obj/local/armeabi-v7a
+directory /home/radmin/Downloads/android-ndk-r15c/sysroot /home/radmin/Downloads/android-ndk-r15c/sources/android/support /home/radmin/Downloads/android-ndk-r15c/sources/android/support/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++ /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/include /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/libjpeg-turbo-1.4.1/simd /home/radmin/Documents/Public_Git_Projects/minicap/jni/minicap-shared/aosp/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi/include jni/libpng
```
Comment (koral--) [inline libs/armeabi-v7a/gdb.setup:2]: Remove this file

File: libs/x86/gdb.setup [hunk 3]
```diff
@@ -0,0 +1,2 @@
+set solib-search-path ./obj/local/x86
+directory /home/radmin/Downloads/android-ndk-r15c/sysroot /home/radmin/Downloads/android-ndk-r15c/sources/android/support /home/radmin/Downloads/android-ndk-r15c/sources/android/support/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++ /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/include /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/libjpeg-turbo-1.4.1/simd /home/radmin/Documents/Public_Git_Projects/minicap/jni/minicap-shared/aosp/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi/include jni/libpng
```
Comment (koral--) [inline libs/x86/gdb.setup:2]: Remove this file

File: libs/x86_64/gdb.setup [hunk 4]
```diff
@@ -0,0 +1,2 @@
+set solib-search-path ./obj/local/x86_64
+directory /home/radmin/Downloads/android-ndk-r15c/sysroot /home/radmin/Downloads/android-ndk-r15c/sources/android/support /home/radmin/Downloads/android-ndk-r15c/sources/android/support/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++ /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/include /home/radmin/Documents/Public_Git_Projects/minicap/jni/libjpeg-turbo/libjpeg-turbo-1.4.1/simd /home/radmin/Documents/Public_Git_Projects/minicap/jni/minicap-shared/aosp/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++/include /home/radmin/Downloads/android-ndk-r15c/sources/cxx-stl/llvm-libc++abi/include jni/libpng
```
Comment (koral--) [inline libs/x86_64/gdb.setup:2]: Remove this file

Comment (varundtsfi) [reply libs/x86_64/gdb.setup:2]: Oh! All removed from each architecture.

**Review by koral-- — APPROVED** [review]

**Comment by ZhouYixun** [comment]
@varundtsfi  Sorry, but how to compile android15 files?

```
Android NDK: android-35 is above the maximum supported version android-31. Choose a supported API level or set APP_PLATFORM to "latest".
D:/android/ndk-bundle/build//../build/core/setup-app-platform.mk:101: *** Android NDK: Aborting.    .  Stop.
```

I get the errors

**Comment by varundtsfi** [comment]
> @varundtsfi Sorry, but how to compile android15 files?
> 
> ```
> Android NDK: android-35 is above the maximum supported version android-31. Choose a supported API level or set APP_PLATFORM to "latest".
> D:/android/ndk-bundle/build//../build/core/setup-app-platform.mk:101: *** Android NDK: Aborting.    .  Stop.
> ```
> 
> I get the errors

Hi @ZhouYixun ,

Today I have downloaded the Android 15  AOSP and I am able to compile and configure it.
**When you are getting this problem while configuring**

make -j4
============================================
PLATFORM_VERSION_CODENAME=REL
PLATFORM_VERSION=15
TARGET_PRODUCT=aosp_arm64
TARGET_BUILD_VARIANT=user
TARGET_ARCH=arm64
TARGET_ARCH_VARIANT=armv8-a
TARGET_CPU_VARIANT=generic
TARGET_2ND_ARCH=arm
TARGET_2ND_ARCH_VARIANT=armv8-a
TARGET_2ND_CPU_VARIANT=generic
HOST_OS=linux
HOST_OS_EXTRA=Linux-6.8.0-40-generic-x86_64-Ubuntu-22.04.4-LTS
HOST_CROSS_OS=windows
BUILD_ID=AP3A.240905.015.A2
OUT_DIR=out

or while running the **lunch** command after configuring the **make** command?
I have the following system information:- 

**Ubuntu 22.04.4 LTS**
**NDK android-ndk-r12b**


**Note:- @ZhouYixun  Can you create a separate issue for this as its related with minicap.so.** 
Screenshot from 2024-09-06 14-47-23

**Comment by ZhouYixun** [comment]
> > @varundtsfi Sorry, but how to compile android15 files?
> > ```
> > Android NDK: android-35 is above the maximum supported version android-31. Choose a supported API level or set APP_PLATFORM to "latest".
> > D:/android/ndk-bundle/build//../build/core/setup-app-platform.mk:101: *** Android NDK: Aborting.    .  Stop.
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I get the errors
> 
> Hi @ZhouYixun ,
> 
> Today I have downloaded the Android 15 AOSP and I am able to compile and configure it. **When you are getting this problem while configuring**
> 
> # make -j4
> PLATFORM_VERSION_CODENAME=REL PLATFORM_VERSION=15 TARGET_PRODUCT=aosp_arm64 TARGET_BUILD_VARIANT=user TARGET_ARCH=arm64 TARGET_ARCH_VARIANT=armv8-a TARGET_CPU_VARIANT=generic TARGET_2ND_ARCH=arm TARGET_2ND_ARCH_VARIANT=armv8-a TARGET_2ND_CPU_VARIANT=generic HOST_OS=linux HOST_OS_EXTRA=Linux-6.8.0-40-generic-x86_64-Ubuntu-22.04.4-LTS HOST_CROSS_OS=windows BUILD_ID=AP3A.240905.015.A2 OUT_DIR=out
> 
> or while running the **lunch** command after configuring the **make** command? I have the following system information:-
> 
> **Ubuntu 22.04.4 LTS** **NDK android-ndk-r12b** I guess you need to downgrade you ndk
> 
> **Note:- @ZhouYixun Can you create a separate issue for this as its related with minicap.so.** Screenshot from 2024-09-06 14-47-23

I still failed... could you upload your Android 15 minicap.so to your resp like https://github.com/varundtsfi/Android-sdk-32-minicap?

**Pull request merged**